base/special/trig.jl

```diff
@@ -1073,6 +1073,26 @@ end
 
 tand(x::Real) = sind(x) / cosd(x)
 
+"""
+    sincosd(x)
+
+Simultaneously compute the sine and cosine of `x`, where `x` is in degrees.
+
+"""
+@inline function sincosd(x::Real)
+    if isinf(x)
+        return throw(DomainError(x, "sincosd(x) is only defined for finite `x`."))
+    elseif isnan(x)
+        return (oftype(x,NaN), oftype(x,NaN))
+    end
+
+    # It turns out that calling those functions separately yielded better
```

Comment:
This is pretty surprising, any idea why?  Part of the point of sincos, is it avoid the double cost of calling sin and cos on the same argument.

Reply:
I think the compiler is clever enough to remove those duplicate executions.

Reply:
In fact, the `sincos_kernel` use the same approach and it has a comment saying that inlining takes care of duplicated computations:

https://github.com/JuliaLang/julia/blob/8221c87b09b8456f4fe422ee2d8756a2a2630c7b/base/special/trig.jl#L208